packages/gitbook/src/middleware.ts

```diff
@@ -259,12 +261,12 @@ export async function middleware(request: NextRequest) {
     }
 
     const customization = url.searchParams.get('customization');
-    if (customization) {
+    if (customization && validateSerializedCustomization(customization)) {
```

Comment:
What is the reason we need to validate this? We send it from our gitbook app and if it's broken then the problem is there not here. We are just suppressing it here

Reply:
https://gitbook-x.sentry.io/issues/6216219929/events/a6aa7a54b061407d8c1e7ce7aa57ed7e/

We can't pass any value in the header.

Reply:
So everything coming from query string should be validated.

Reply:
See https://dyor.hacken.ai/~gitbook/icon?size=small&theme=1ACUSTART%27%22*%2F%0D%0A+%3C%3FphpACUEND

Reply:
Right but how is this `theme` being passed? Again, we generate these URLs and if the values are garbled then we need to fix the problem at source.

Reply:
If it's public it has to be validated even if we are generating the URL anything can modify the query string or put one.

Reply:
Okay so users are passing garbage values themselves

Reply:
Exactly, especially for themes. Internet and browsers are wild!